code/modules/atmospherics/machinery/components/machines/crystallizer.dm

```diff
@@ -0,0 +1,332 @@
+	icon = 'icons/obj/atmospherics/components/machines.dmi'
+	icon_state = "crystallizer-off"
+	name = "crystallizer"
+	desc = "Used to crystallize or solidify gases."
+	layer = ABOVE_MOB_LAYER
+	density = TRUE
+	max_integrity = 300
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 100, BOMB = 0, BIO = 100, RAD = 100, FIRE = 80, ACID = 30)
+	circuit = /obj/item/circuitboard/machine/crystallizer
+	pipe_flags = PIPING_ONE_PER_TURF
+
+	///Base icon state for the machine to be used in update_icon()
+	var/base_icon = "crystallizer"
+	///Internal Gas mix used for processing the gases that have been put in
+	var/datum/gas_mixture/internal
+	///Var that controls how much gas gets injected
+	var/gas_input = 0
+	///Var that checks if the gases put in are the same numbers as the requirements
+	var/gas_check = 0
+	///Saves the progress during the processing of the items
+	var/progress_bar = 0
+	///Stores the amount of lost quality
+	var/quality_loss = 0
+	///Stores the recipe selected by the user in the GUI
+	var/recipe_type = null
+	///Stores the total amount of moles needed for the current recipe
+	var/total_recipe_moles = 0
+
+/obj/machinery/atmospherics/components/binary/crystallizer/Initialize()
+	. = ..()
+	internal = new(2000)
+
+/obj/machinery/atmospherics/components/binary/crystallizer/attackby(obj/item/I, mob/user, params)
+	if(!on)
+		if(default_deconstruction_screwdriver(user, "[base_icon]-open", "[base_icon]-off", I))
+			return
+	if(default_change_direction_wrench(user, I))
+		return
+	if(default_deconstruction_crowbar(I))
+		return
+	return ..()
+
+/obj/machinery/atmospherics/components/binary/crystallizer/default_change_direction_wrench(mob/user, obj/item/I)
+	. = ..()
+	if(!.)
+		return FALSE
+	SetInitDirections()
+	var/obj/machinery/atmospherics/node1 = nodes[1]
+	var/obj/machinery/atmospherics/node2 = nodes[2]
+	if(node1)
+		if(src in node1.nodes) //Only if it's actually connected. On-pipe version would is one-sided.
+			node1.disconnect(src)
+		nodes[1] = null
+	if(node2)
+		if(src in node2.nodes) //Only if it's actually connected. On-pipe version would is one-sided.
+			node2.disconnect(src)
+		nodes[2] = null
+
+	if(parents[1])
+		nullifyPipenet(parents[1])
+	if(parents[2])
+		nullifyPipenet(parents[2])
+
+	atmosinit()
+	node1 = nodes[1]
+	if(node1)
+		node1.atmosinit()
+		node1.addMember(src)
+	node2 = nodes[2]
+	if(node2)
+		node2.atmosinit()
+		node2.addMember(src)
+	SSair.add_to_rebuild_queue(src)
+	return TRUE
+
+/obj/machinery/atmospherics/components/binary/crystallizer/update_icon()
+	. = ..()
+	cut_overlays()
+
+	if(panel_open)
+		icon_state = "[base_icon]-open"
+	else if(on && is_operational)
+		icon_state = "[base_icon]-on"
+	else
+		icon_state = "[base_icon]-off"
+
+	add_overlay(getpipeimage(icon, "pipe", dir, COLOR_LIME, piping_layer))
+	add_overlay(getpipeimage(icon, "pipe", turn(dir, 180), COLOR_MOSTLY_PURE_RED, piping_layer))
+
+/obj/machinery/atmospherics/components/binary/crystallizer/CtrlClick(mob/living/user)
```

Comment:
Might be worth considering moving this to rightclick now.

Reply:
yes i was thinking of changing all other atmos machines that uses ctrl click to r-click, so i'll do that here too